Redis 0.20.1 with tokio-comp feature failed to build on CentOS7

OS: CentOS7
Rust: 1.54.0
Cargo: 1.54.0
Cargo deps: redis = { version = "0.20.1", features = ["tokio-comp"] }

```
error[E0412]: cannot find type `BoxFuture` in module `futures_util::future`
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/types.rs:529:53
    |
529 |   pub type RedisFuture<'a, T> = futures_util::future::BoxFuture<'a, RedisResult<T>>;
    |                                                       ^^^^^^^^^ help: a trait with a similar name exists: `Future`

error[E0599]: no method named `boxed` found for opaque type `impl futures_util::Future` in the current scope
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:531:10
    |
531 |         .boxed()
    |          ^^^^^ method not found in `impl futures_util::Future`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `RedisRuntime` defines an item `boxed`, perhaps you need to implement it
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:56:1
    |
56  | pub(crate) trait RedisRuntime: AsyncStream + Send + Sync + Sized + 'static {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `boxed` found for opaque type `impl futures_util::Future` in the current scope
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:581:10
    |
581 |         .boxed()
    |          ^^^^^ method not found in `impl futures_util::Future`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `RedisRuntime` defines an item `boxed`, perhaps you need to implement it
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:56:1
    |
56  | pub(crate) trait RedisRuntime: AsyncStream + Send + Sync + Sized + 'static {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `boxed` found for opaque type `impl futures_util::Future` in the current scope
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:895:10
    |
895 |         .boxed()
    |          ^^^^^ method not found in `impl futures_util::Future`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `RedisRuntime` defines an item `boxed`, perhaps you need to implement it
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:56:1
    |
56  | pub(crate) trait RedisRuntime: AsyncStream + Send + Sync + Sized + 'static {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no method named `boxed` found for opaque type `impl futures_util::Future` in the current scope
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:918:10
    |
918 |         .boxed()
    |          ^^^^^ method not found in `impl futures_util::Future`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `RedisRuntime` defines an item `boxed`, perhaps you need to implement it
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/redis-0.20.2/src/aio.rs:56:1
    |
56  | pub(crate) trait RedisRuntime: AsyncStream + Send + Sync + Sized + 'static {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

   Compiling hyper v0.14.12
error: aborting due to 5 previous errors

Some errors have detailed explanations: E0412, E0599.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `redis`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed

```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,7 +61,7 @@ async-native-tls = { version = "0.3", optional = true }
 [features]
 default = ["acl", "streams", "geospatial", "script"]
 acl = []
-aio = ["bytes", "pin-project-lite", "futures-util", "futures-util/sink", "tokio/io-util", "tokio-util", "tokio-util/codec", "tokio/sync", "combine/tokio"]
+aio = ["bytes", "pin-project-lite", "futures-util", "futures-util/alloc", "futures-util/sink", "tokio/io-util", "tokio-util", "tokio-util/codec", "tokio/sync", "combine/tokio"]
 geospatial = []
 cluster = ["crc16", "rand"]
 script = ["sha1"]

--- a/tests/test_cluster.rs
+++ b/tests/test_cluster.rs
@@ -190,6 +190,7 @@ fn test_cluster_pipeline_command_ordering() {
 }
 
 #[test]
+#[ignore] // Flaky
 fn test_cluster_pipeline_ordering_with_improper_command() {
     let cluster = TestClusterContext::new(3, 0);
     cluster.wait_for_cluster_up();

